src/modules/did/did.service.spec.ts

```diff
@@ -1,19 +1,109 @@
+const MockLogger = {
+  log: jest.fn(),
+  info: jest.fn(),
+  debug: jest.fn(),
+  setContext: jest.fn(),
+};
+const MockObject = {};
+const MockConfigService = {
+  get: jest.fn((key: string) => {
+    if (key === 'DID_SYNC_ENABLED') {
+      return 'false';
+    }
+    return null;
+  }),
+};
+let didDocEntity: DIDDocumentEntity;
+let didDoc: IDIDDocument
+const repositoryMockFactory = jest.fn(() => ({
+  findOne: jest.fn(() => {
+    return didDocEntity;
+  }),
+  save: jest.fn(entity => entity)
+}));
+const queueMockFactory = jest.fn(() => ({
+}));
+jest.mock('@ew-did-registry/did-ipfs-store', () => {
+  return {
+    DidStore: jest.fn(() => { return {} })
+  }
+});
+jest.mock('@ew-did-registry/did-ethr-resolver', () => ({
+  ...jest.requireActual('@ew-did-registry/did-ethr-resolver') as Record<string, unknown>,
+  documentFromLogs: jest.fn(() => didDoc),
+  Resolver: jest.fn(() => {
+    return {
+      readFromBlock: jest.fn(() => "<logs>")
+    }
+  })
+}));
+jest.mock('../../ethers/EthereumDidRegistryFactory', () => ({
+  EthereumDidRegistryFactory: {
+    connect: jest.fn(() => {
+      return {
+        addEventListener: jest.fn()
+      }
+    })
+  }
+}));
+
+describe('DidDocumentService', () => {
   let service: DIDService;
 
   beforeEach(async () => {
     const module: TestingModule = await Test.createTestingModule({
-      providers: [DIDService],
+      providers: [DIDService,
+        { provide: ConfigService, useValue: MockConfigService },
+        { provide: SchedulerRegistry, useValue: MockObject },
+        { provide: HttpService, useValue: MockObject },
+        { provide: Logger, useValue: MockLogger },
+        { provide: getQueueToken('dids'), useFactory: queueMockFactory },
+        { provide: getRepositoryToken(DIDDocumentEntity), useFactory: repositoryMockFactory },
+        { provide: Provider, useValue: MockObject }
+      ]
     }).compile();
 
+    didDoc = {
+      id: "<id>",
+      service: [],
+      authentication: [],
+      publicKey: [],
+      created: "<created>",
+      proof: undefined,
+      updated: "<updated>",
+      '@context': "<context>"
+    };
+    didDocEntity = Object.assign(didDoc, { logs: "<logs>" })
     service = module.get<DIDService>(DIDService);
   });
 
   it('should be defined', () => {
     expect(service).toBeDefined();
   });
+
+  // Would be better done by exact types: https://github.com/Microsoft/TypeScript/issues/12936
+  it('getByID should return only IDIDDocument params from cached document', async () => {
+    const didDoc = await service.getById("<any DID>");
+    expect(didDoc).toBeDefined();
+    expect(didDoc[nameof<DIDDocumentEntity>("logs")]).toBeUndefined();
+  });
+
+  it('getByID should return only IDIDDocument params from non-cached document', async () => {
+    didDocEntity = undefined;
+    const didDoc = await service.getById("<any DID>");
+    expect(didDoc).toBeDefined();
+    expect(didDoc[nameof<DIDDocumentEntity>("logs")]).toBeUndefined();
```

Comment:
IMO  methods like `toBeDefined`, `toBeUndefined` don't help that much when it comes to ensuring deterministic tests. I think we should also assert for the actual values we expect and probably mock methods that get called. this would help catching type  related bugs .

Reply:
Thanks. Refactored test to check for each property